src/TenorGuitar/Validation.purs

```diff
@@ -0,0 +1,138 @@
+module TenorGuitar.Validation (validate, validateJson) where
+
+-- | Validate a TenorGuitar chord pattern when loaded from a JSON file
+
+import Prelude (($), (>), (<), (>=), (<=), (||), (&&), (<>), (<$>), (<*>), (/=), (-), const, map, negate, pure, show)
+import Data.Array (filter, length, mapWithIndex)
+import Data.Foldable (intercalate)
+import Data.Either (either)
+import Data.Maybe (Maybe(..))
+import Data.Validation.Semigroup
+import TenorGuitar.Types
+  ( ChordShape
+  , Barre
+  , FingeredString
+  , FingerPosition
+  , Fingering
+  , displayedFretCount
+  )
+import Common.Types (Validated)
+import Serialization.Json (readTenorGuitar)
+
+maxFrets :: Int
+maxFrets = 20
+
+-- | validate a prospective tenor guitar chord JSON String
+validateJson :: String -> Validated ChordShape
+validateJson json =
+  either
+    (const $ invalid $ pure "Not a recognisable tenor guitar chord format.")
+    validate
+    (readTenorGuitar json)
+
+-- | validate a prospective tenor guitar chord
+validate :: ChordShape -> Validated ChordShape
+validate chordShape =
+  ( { name: chordShape.name, firstFretOffset: _, barre: _, fingering: _ }
+      <$> validateFirstFretOffset chordShape.firstFretOffset
+      <*> validateBarre chordShape.barre
+      <*> validateFingering chordShape.fingering
+  ) `andThen` checkHiddenByBarre
+
+validateFirstFretOffset :: Int -> Validated Int
+validateFirstFretOffset offset =
+  if (offset < 0) || (offset > maxFrets) then
+    invalid $ pure ("First fret offset should be between 0 and " <> show maxFrets <> ".")
+  else
+    pure offset
+
+validateBarre :: Barre -> Validated Barre
+validateBarre mbarre =
+  case mbarre of
+    Just barre ->
+      if (barre.stringNumber < 0 || barre.stringNumber >= 6) then
+        invalid $ pure
+          ( "Invalid string number of "
+              <> show barre.stringNumber
+              <> " in the barré."
+          )
+      else if (barre.fretNumber < 1 || barre.fretNumber >= displayedFretCount) then
+        invalid $ pure
+          ( "Invalid fret number of "
+              <> show barre.fretNumber
+              <> " in the barré which should be between 1 and "
+              <> show (displayedFretCount - 1)
+              <> "."
+          )
+      else
+        pure mbarre
+    _ ->
+      pure mbarre
+
+validateFingering :: Fingering -> Validated Fingering
+validateFingering fingering =
+  if (length fingering /= 6) then
```

Comment:
This should be 4. This should not really be a magic number but there should be a constant declared.  I suggest we use 4 for the moment, but when we come to  refactor, we somehow gather together all the common constants for string instruments into a data structure/

Reply:
Sounds good to me :() I added your request in 1fb777f886ffede1da1d542423614bd010d24a5d